lib/rules/selector-max-id/__tests__/index.js

```diff
@@ -167,6 +167,26 @@ testRule(rule, {
     {
       code: "#foo { @media print { #bar {} } }",
       description: "media query: nested"
+    },
+    {
+      code: "#foo { @if ($p == 1) { #bar {} } }",
+      description: "@if statement: nested"
+    },
+    {
+      code: ".foo { @if ($p == 1) { #bar #baz {} } }",
+      description: "@if statement: nested compound selector"
+    },
```

Comment:
I think tests containing `@if` should go to `test` function with `syntax: "scss"`.